File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/ui/fragments/base/BaseFragment.java
package pl.pjwstk.bmiapp.ui.fragments.base;

import android.os.Bundle;
import android.view.View;
import android.view.ViewTreeObserver;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import lombok.AccessLevel;
import lombok.experimental.FieldDefaults;

@FieldDefaults(level = AccessLevel.PROTECTED)
public abstract class BaseFragment extends Fragment {

    View rootView;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        rootView = view;
    }

    @Override
    public void onResume() {
        super.onResume();
        fixLayout();
    }

    public void refreshLayout() {
        fixLayout();
    }

    protected void fixLayout() {
        if (rootView != null) {
            rootView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
                @Override
                public void onGlobalLayout() {
                    rootView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
                    rootView.requestLayout();
                }
            });
        }
    }
}

File: BMIApp/app/src/main/java/pl/pjwstk/bmiapp/ui/fragments/base/BaseFragment.kt
package pl.pjwstk.bmiapp.ui.fragments.base

import android.os.Bundle
import android.view.View
import android.view.ViewTreeObserver
import androidx.fragment.app.Fragment

abstract class BaseFragment : Fragment() {

    protected var rootView: View? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        rootView = view
    }

    override fun onResume() {
        super.onResume()
        fixLayout()
    }

    fun refreshLayout() {
        fixLayout()
    }

    protected open fun fixLayout() {
        rootView?.let { root ->
            root.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
                override fun onGlobalLayout() {
                    root.viewTreeObserver.removeOnGlobalLayoutListener(this)
                    root.requestLayout()
                }
            })
        }
    }
}